index/scorch/snapshot_rollback.go

```diff
@@ -0,0 +1,183 @@
+//  Copyright (c) 2017 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorch
+
+import (
+	"bytes"
+	"log"
+
+	"github.com/blevesearch/bleve/index/scorch/segment"
+)
+
+type indexSnapErr struct {
+	i *IndexSnapshot
+	e error
+}
+
+// Snapshot fetches the latest persisted IndexSnapshot.
+func (s *Scorch) Snapshot() (*IndexSnapshot, error) {
```

Comment:
The method name here is probably too general -- easy to inadvertently entice folks into thinking that they're getting the latest snapshot?  Perhaps LastPersistedSnapshot() is a clearer name of what the method's doing?

Reply:
Done.